adapters/oidc/js/rollup.config.ts

```diff
@@ -0,0 +1,81 @@
+import commonjs from "@rollup/plugin-commonjs";
+import inject from "@rollup/plugin-inject";
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import path from "node:path";
+import type { OutputOptions, RollupOptions } from "rollup";
+import { defineConfig } from "rollup";
+import { terser } from "rollup-plugin-terser";
+
+interface DefineOptionsArgs {
+  file: string;
+  name: string;
+  amdId: string;
+}
+
+function defineOptions({
+  file,
+  name,
+  amdId,
+}: DefineOptionsArgs): RollupOptions[] {
+  const sourceDir = "src/main/js";
+  const targetDir = "target/classes";
+  const commonOptions: RollupOptions = {
+    input: path.join(sourceDir, `${file}.js`),
+    plugins: [commonjs(), nodeResolve()],
+  };
+
+  const umdOutput: OutputOptions = {
+    format: "umd",
+    name,
+    amd: { id: amdId },
+  };
+
+  return [
+    // Modern ES module variant, with externalized dependencies.
+    {
+      ...commonOptions,
+      output: [
+        {
+          file: path.join(targetDir, `${file}.mjs`),
+        },
+      ],
+      external: ["base64-js", "js-sha256"],
+    },
+    // Legacy Universal Module Definition, or “UMD”, with inlined dependencies.
+    {
+      ...commonOptions,
+      output: [
+        {
+          ...umdOutput,
+          file: path.join(targetDir, `${file}.js`),
+        },
+        {
+          ...umdOutput,
+          file: path.join(targetDir, `${file}.min.js`),
+          sourcemap: true,
+          sourcemapExcludeSources: true,
+          plugins: [terser()],
+        },
+      ],
+      plugins: [
+        ...commonOptions.plugins,
+        inject({
+          Promise: ["es6-promise/dist/es6-promise.min.js", "Promise"],
```

Comment:
If I understand correctly this is a replacement for the `promise-polyfill` we had hardcoded inside `keycloak.js` previously. This is inlined into the resulting `keycloak.js` so the `keycloak.js` file available from `/auth/js/keycloak.js` always has the Promise definition. Is it correct?

However, when `keycloak-js` is used from npm (`keycloak.mjs`), the `Promise` definition seems to be missing. Are we are relying on users of `keycloak-js` to import something that provides Promise themselves, otherwise, it throws an exception [1]? Does it mean that by default applications using `keycloak-js` adapter do not work on old browsers?

[1]
```
throw Error('Keycloak requires an environment that supports Promises. Make sure that you include the appropriate polyfill.');
```

Reply:
Yes that is correct, this is indeed intended to ensure the scripts that are served statically from the Keycloak server always include a polyfill for `Promise`. Since these statically served scripts are unversioned we have to assume the worst case scenario in terms of browser compatibility, as the server might get upgraded and still serve older browsers in application code.

In the case of the ES module variant we are indeed not loading the Promise polyfill by default to reduce final bundle size for end users. The reason for this is that any browser (or other runtime) that supports ES modules natively also has support for `Promise`.

This still leaves the edge case of build tooling that bundles this kind of module into an older variant of JavaScript (such as ES5). However since the support for the `module` field is quite new in front-end tooling, I think we can safely assume that users that make use of these tools and want to support older browsers are already actively shipping a polyfill for `Promise`.

For those very small subset of users that do not meet the aforementioned conditions, they would still be protected by the fact that they need to upgrade to a new major version (Keycloak 17) of the package. I've made the choice here not to burden new applications with an increased bundle size vs. supporting a group of users that likely does not exist.

Reply:
I just realized the last edge case actually does not exist either due to the fact that we were already conditionally loading the polyfill in the [module detection code](https://github.com/keycloak/keycloak/pull/9351/files#diff-eeb68cbf057af16d2dec09ea923a38332685546740f43d5979d758b39181bd89L25).

Reply:
Ahh, I see that. Thank you very much for the explanation!